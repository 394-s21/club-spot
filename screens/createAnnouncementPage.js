import { Component } from 'react';
import React from "react";
import { SafeAreaView, StyleSheet, View, ScrollView, Alert} from 'react-native';
import { Provider, TextInput, RadioButton,Text, Subheading,Card, Button,Paragraph, Dialog, Portal } from 'react-native-paper';
import { firebase }  from '../utils/firebase';

class createAnnouncementPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      clubId : this.props.route.params.clubId,
      clubName : this.props.route.params.clubName,
      announcement : "",
    }
  }
  
  alertUser(title, subtitle) {
    Alert.alert(
      title,
      subtitle,
      [
          { text: "OK" }
      ]
    );
  }
  createAnnouncement = () => {
    let data = {
      clubId : this.state.clubId,
      clubName: this.state.clubName,
      announcement: this.state.announcement
    }
    console.log("data is ", data)
    const db = firebase.database().ref()
    db.child('/announcements/' + data.clubId).set(data)
    this.alertUser("Update Successful", "")
    this.props.navigation.pop() // go back to the club announcement page
  }
  render() {
    return(
      <View>
        <TextInput 
          label='What is your club announcement?' 
          value={this.state.announcement} 
          type="outlined"
          
          style={styles.field}
          onChangeText={text => this.setState({announcement:text})} />
        <Button style={styles.button} mode="outlined" onPress = {this.createAnnouncement} > Create Announcement </Button>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  field: {
    marginTop: 15,
    height: 55,
    padding: 5,
    backgroundColor: 'white',
  },
  button: {
    marginTop: 20
  },
})
export default createAnnouncementPage