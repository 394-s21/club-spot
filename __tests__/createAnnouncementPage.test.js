import React from  'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { View } from 'react-native';
import { Card } from 'react-native-paper';
import createAnnouncementPage from '../screens/createAnnouncementPage';


test('renders createAnnouncementPage correctly', () => {
    const page = renderer.create(<createAnnouncementPage/>).toJSON();
    expect(page).toMatchSnapshot();

});