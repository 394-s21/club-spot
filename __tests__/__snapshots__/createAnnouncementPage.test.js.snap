// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders createAnnouncementPage correctly 1`] = `<createAnnouncementPage />`;
